client/engine/chainservice/eth_chainservice.go

```diff
@@ -44,24 +44,32 @@ func NewEthChainService(chain ethChain, na *NitroAdjudicator.NitroAdjudicator, n
 	return &ecs
 }
 
-// SendTransaction sends the transaction and blocks until it has been submitted.
-func (ecs *EthChainService) SendTransaction(tx protocols.ChainTransaction) *ethTypes.Transaction {
-	txOpt := bind.TransactOpts{
+// defaultTxOpts return transaction options suitable for most transaction submissions
+func (ecs *EthChainService) defaultTxOpts() bind.TransactOpts {
+	return bind.TransactOpts{
 		From:     ecs.txSigner.From,
 		Nonce:    ecs.txSigner.Nonce,
 		Signer:   ecs.txSigner.Signer,
 		GasPrice: big.NewInt(10000000000),
 	}
+}
+
+// SendTransaction sends the transaction and blocks until it has been submitted.
+func (ecs *EthChainService) SendTransaction(tx protocols.ChainTransaction) *ethTypes.Transaction {
 	switch tx := tx.(type) {
 	case protocols.DepositTransaction:
 		for tokenAddress, amount := range tx.Deposit {
-			txOpt.Value = amount
+			txOpts := ecs.defaultTxOpts()
+			ethTokenAddress := common.Address{}
```

Comment:
Should this be extracted somewhere more central? We rely on it in many places...

Reply:
Good call, filed https://github.com/statechannels/go-nitro/issues/766.